docker_auth.go

```diff
@@ -80,24 +95,93 @@ func defaultRegistry(ctx context.Context) string {
 	return info.IndexServerAddress
 }
 
-// authConfig represents the details of the auth config for a registry.
-type authConfig struct {
+// authConfigResult is a result looking up auth details for key.
+type authConfigResult struct {
 	key string
 	cfg registry.AuthConfig
+	err error
+}
+
+// credentialsCache is a cache for registry credentials.
```

Comment:
In v1 there will be an `auth` package in the root level. Do you think we can move this struct and methods to `internal/core/auth` so the forward-port is easier? 🤔

Reply:
I'd have to look into this some more, would suggest we move at a later date if we do.